src/backend/dbi/gnc-backend-dbi.c

```diff
@@ -788,6 +788,137 @@ gnc_dbi_unlock( QofBackend *qbe )
     qof_backend_set_error( qbe, ERR_BACKEND_SERVER_ERR );
 }
 
+#define SQL_OPTION_TO_REMOVE "NO_ZERO_DATE"
+
+// Given an sql_options string returns a copy of the string adjusted as necessary.
+// In particular if the string contains SQL_OPTION_TO_REMOVE it is removed along with 
+// comma separator.
+static gchar*
+adjust_sql_options_string( const gchar *str )
+{
+    char* pos;
+    char* answer = g_malloc( strlen(str)+1 );  // this must be freed by the calling code
+    int chars_to_copy;
+    PERR( "\nIn %s", str );
+    pos = strstr( str, SQL_OPTION_TO_REMOVE );
+    if (pos)
+    {
+        PERR("%c %c", *(pos-1), *(pos+(strlen(SQL_OPTION_TO_REMOVE)+1)));
+        if ( pos == str )
+        {
+            PERR( "A" );
+            if ( *(pos+strlen(SQL_OPTION_TO_REMOVE)) == ',' )
+            {
+            PERR( "B" );
+                strcpy( answer, str + strlen(SQL_OPTION_TO_REMOVE) + 1 );
+            } else if ( *(pos+strlen(SQL_OPTION_TO_REMOVE)) == '\0' )
+            {
+            PERR( "C" );
+                strcpy( answer, "" );
+            } else
+            {
+                PERR( "A1" );
+                strcpy( answer, str );
+            }
+        } else if ( *(pos + strlen(SQL_OPTION_TO_REMOVE)) == '\0' )
+        {
+            PERR( "D" );
+            chars_to_copy =  strlen(str) - strlen(SQL_OPTION_TO_REMOVE) - 1;
+            strncpy( answer, str, chars_to_copy );
+            answer[chars_to_copy] = '\0';
+        } else if (*(pos-1) == ','  &&  *(pos+strlen(SQL_OPTION_TO_REMOVE)) == ',')
+        {
+            PERR( "E" );
+            chars_to_copy =  pos - str - 1;
+            strncpy( answer, str, chars_to_copy );
+            strcpy( answer + chars_to_copy, pos+strlen(SQL_OPTION_TO_REMOVE));
+        } else
+        {
+            PERR( "F" );
+            // not found
+            strcpy( answer, str );
+        }
+    } else
+    {
+        PERR( "G" );
+        // not found
+        strcpy( answer, str );
+    }
+    PERR( "Out %s\n", answer );
+    return answer;
+}
```

Comment:
The regex as proposed catches all of these:
NO_ZERO_DATE (ie when it's the full string)
NO_ZERO_DATE, (ie when the option is first in the list)
,NO_ZERO_DATE (ie when the option is at the end of the list)
,NO_ZERO_DATE, (ie when the option is somewhere in the middle of the list)

All of these would be replaced with an empty string, which is indeed not correct for the last case.

How about this: "(,?NO_ZERO_DATE|NO_ZERO_DATE,?)"

That would catch
NO_ZERO_DATE
  -> when it's the full string
,NO_ZERO_DATE
  -> when the option is at the end of the list
  -> or when the option is somewhere in the middle, but then will leave the trailing "," to keep the list properly separated
NO_ZERO_DATE, 
  -> when the option is first in the list